jme3-core/src/main/java/com/jme3/environment/baker/GenericEnvBaker.java

```diff
@@ -0,0 +1,254 @@
+ *
+ * @author Riccardo Balbo
+ */
+public abstract class GenericEnvBaker implements EnvBaker {
+
+    protected static Vector3f[] axisX = new Vector3f[6];
+    protected static Vector3f[] axisY = new Vector3f[6];
+    protected static Vector3f[] axisZ = new Vector3f[6];
+    static {
+        // PositiveX axis(left, up, direction)
+        axisX[0] = Vector3f.UNIT_Z.mult(1.0F);
+        axisY[0] = Vector3f.UNIT_Y.mult(-1.0F);
+        axisZ[0] = Vector3f.UNIT_X.mult(1.0F);
+        // NegativeX
+        axisX[1] = Vector3f.UNIT_Z.mult(-1.0F);
+        axisY[1] = Vector3f.UNIT_Y.mult(-1.0F);
+        axisZ[1] = Vector3f.UNIT_X.mult(-1.0F);
+        // PositiveY
+        axisX[2] = Vector3f.UNIT_X.mult(-1.0F);
+        axisY[2] = Vector3f.UNIT_Z.mult(1.0F);
+        axisZ[2] = Vector3f.UNIT_Y.mult(1.0F);
+        // NegativeY
+        axisX[3] = Vector3f.UNIT_X.mult(-1.0F);
+        axisY[3] = Vector3f.UNIT_Z.mult(-1.0F);
+        axisZ[3] = Vector3f.UNIT_Y.mult(-1.0F);
+        // PositiveZ
+        axisX[4] = Vector3f.UNIT_X.mult(-1.0F);
+        axisY[4] = Vector3f.UNIT_Y.mult(-1.0F);
+        axisZ[4] = Vector3f.UNIT_Z;
+        // NegativeZ
+        axisX[5] = Vector3f.UNIT_X.mult(1.0F);
+        axisY[5] = Vector3f.UNIT_Y.mult(-1.0F);
+        axisZ[5] = Vector3f.UNIT_Z.mult(-1.0F);
+    }
+
+    protected TextureCubeMap env;
+    protected Format depthFormat;
+
+    protected final RenderManager renderManager;
+    protected final AssetManager assetManager;
+    protected final Camera cam;
+    protected  boolean texturePulling=false;
+    protected List<ByteArrayOutputStream> bos = new ArrayList<>();
+    private static final Logger LOG=Logger.getLogger(GenericEnvBaker.class.getName());
+
+ 
+
+    public GenericEnvBaker(RenderManager rm, AssetManager am, Format colorFormat, Format depthFormat, int env_size) {
+        this.depthFormat = depthFormat;
+
+        renderManager = rm;
+        assetManager = am;
+
+        cam = new Camera(128, 128);
+
+        env = new TextureCubeMap(env_size, env_size, colorFormat);
+        env.setMagFilter(MagFilter.Bilinear);
+        env.setMinFilter(MinFilter.BilinearNoMipMaps);
+        env.setWrap(WrapMode.EdgeClamp);
+        env.getImage().setColorSpace(ColorSpace.Linear);
+    }
+
+    @Override
+    public void setTexturePulling(boolean v) {
+        texturePulling = v;
+    }
+
+    @Override
+    public boolean isTexturePulling() {
+        return texturePulling;
+    }
+
+    public TextureCubeMap getEnvMap() {
+        return env;
+    }
+
+    Camera getCam(int id, int w, int h, Vector3f position, float frustumNear, float frustumFar) {
+        cam.resize(w, h, false);
+        cam.setLocation(position);
+        cam.setFrustumPerspective(90.0F, 1F, frustumNear, frustumFar);
+        cam.setRotation(new Quaternion().fromAxes(axisX[id], axisY[id], axisZ[id]));
+        return cam;
+    }
+
+    @Override
+    public void clean() {
+        env.getImage().dispose();
+        System.gc();
```

Comment:
Is there a specific reason for two `System.gc()` calls in a row, and do we absolutely need these? I generally consider `System.gc()` something to be avoided except when testing/debugging certain behaviors (it can negatively impact performance, for one thing, and it's not a reliable way to force GC since it may be a non-op or ignored at the GC's whim).

Reply:
I've removed this method entirely, since the generated env map can be used after the baker is disposed, having a clean method like that might be counterintuitive as it would destroy the env map unexpectedly